pkg/metrics/metrics.go

```diff
@@ -78,6 +78,16 @@ var (
 	},
 		[]string{"outcome"})
 
+	// EndpointStateCount is the total count of the endpoints in various states.
+	EndpointStateCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: Namespace,
+			Name:      "endpoint_state",
+			Help:      "Count of all endpoints, tagged by different endpoint states",
+		},
+		[]string{"endpoint state"},
```

Comment:
Per the Prometheus naming conventions, it seems label names shouldn't contain spaces:
https://prometheus.io/docs/practices/naming/

So I'd suggest either `endpoint_state` or just `state`.

Reply:
ah.. I guess labels are allowed to have spaces but not label names. Let me change that

Reply:
done. Added special case for cilium health as well. please check